### PR TITLE
Bug 1879878: remove logging when authz header is present but basic auth is not attempted

### PR DESCRIPTION
**Pull request by stlaz**
/assign @sttts

### CODE REVIEW

File: oauthproxy.go [hunk 1]
```diff
@@ -840,7 +840,7 @@ func (p *OAuthProxy) CheckBasicAuth(req *http.Request) (*providers.SessionState,
 	}
 	s := strings.SplitN(auth, " ", 2)
 	if len(s) != 2 || s[0] != "Basic" {
-		return nil, fmt.Errorf("Authorization header does not start with 'Basic', skipping basic authentication")
+		return nil, nil
```
Comment (sttts) [inline oauthproxy.go:843]: add go doc what this func's return values mean. I assume nil is not "accept".

**Comment by stlaz** [comment]
@sttts added the requested godoc

**Pull request merged**